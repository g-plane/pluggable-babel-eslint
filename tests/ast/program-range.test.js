const { parse } = require('../..')

test('empty body', () => {
  const code = '//abc'
  expect(parse(code, { plugins: ['typescript'] })).toMatchSnapshot()
})

test('non-empty body', () => {
  const code = '//abc\nabc'
  expect(parse(code, { plugins: ['typescript'] })).toMatchSnapshot()
})
