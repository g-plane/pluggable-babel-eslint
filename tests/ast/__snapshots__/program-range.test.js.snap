// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty body 1`] = `
Node {
  "body": Array [],
  "comments": Array [
    Object {
      "end": 5,
      "loc": SourceLocation {
        "end": Position {
          "column": 5,
          "line": 1,
        },
        "start": Position {
          "column": 0,
          "line": 1,
        },
      },
      "range": Array [
        0,
        5,
      ],
      "start": 0,
      "type": "Line",
      "value": "abc",
    },
  ],
  "directives": undefined,
  "end": 5,
  "loc": SourceLocation {
    "end": Position {
      "column": 5,
      "line": 1,
    },
    "start": Position {
      "column": 5,
      "line": 1,
    },
  },
  "range": Array [
    5,
    5,
  ],
  "sourceType": "module",
  "start": 5,
  "tokens": Array [],
  "type": "Program",
}
`;

exports[`non-empty body 1`] = `
Node {
  "body": Array [
    Node {
      "_babelType": "ExpressionStatement",
      "end": 9,
      "expression": Node {
        "_babelType": "Identifier",
        "end": 9,
        "loc": SourceLocation {
          "end": Position {
            "column": 3,
            "line": 2,
          },
          "identifierName": "abc",
          "start": Position {
            "column": 0,
            "line": 2,
          },
        },
        "name": "abc",
        "range": Array [
          6,
          9,
        ],
        "start": 6,
        "type": "Identifier",
      },
      "leadingComments": Array [
        Object {
          "end": 5,
          "loc": SourceLocation {
            "end": Position {
              "column": 5,
              "line": 1,
            },
            "start": Position {
              "column": 0,
              "line": 1,
            },
          },
          "range": Array [
            0,
            5,
          ],
          "start": 0,
          "type": "Line",
          "value": "abc",
        },
      ],
      "loc": SourceLocation {
        "end": Position {
          "column": 3,
          "line": 2,
        },
        "start": Position {
          "column": 0,
          "line": 2,
        },
      },
      "range": Array [
        6,
        9,
      ],
      "start": 6,
      "type": "ExpressionStatement",
    },
  ],
  "comments": Array [
    Object {
      "end": 5,
      "loc": SourceLocation {
        "end": Position {
          "column": 5,
          "line": 1,
        },
        "start": Position {
          "column": 0,
          "line": 1,
        },
      },
      "range": Array [
        0,
        5,
      ],
      "start": 0,
      "type": "Line",
      "value": "abc",
    },
  ],
  "directives": undefined,
  "end": 9,
  "loc": SourceLocation {
    "end": Position {
      "column": 3,
      "line": 2,
    },
    "start": Position {
      "column": 0,
      "line": 2,
    },
  },
  "range": Array [
    6,
    9,
  ],
  "sourceType": "module",
  "start": 0,
  "tokens": Array [
    Token {
      "end": 9,
      "loc": SourceLocation {
        "end": Position {
          "column": 3,
          "line": 2,
        },
        "start": Position {
          "column": 0,
          "line": 2,
        },
      },
      "range": Array [
        6,
        9,
      ],
      "start": 6,
      "type": "Identifier",
      "value": "abc",
    },
  ],
  "type": "Program",
}
`;
